### PR TITLE
fix pipe_reader on multi passes

### LINKED ISSUE #6529: pipe_read()第二轮训练读不到数据
第一轮训练正常输出结果
第二轮开始直接报ZeroDivisionError: float division by zero
  File "/home/map/paddle_env/python27-gcc482/lib/python2.7/site-packages/paddle/v2/trainer.py", line 220, in test
    evaluator=evaluator, cost=total_cost / num_samples)
ZeroDivisionError: float division by zero

check发现是api代码中，只在开始的时候调用了一次process = subprocess.Popen(left_cmd.split(" "), bufsize=bufsize, stdout=subprocess.PIPE)
数据读完之后，没用重新打开，所以第二轮训练读不到数据

辛苦paddlepaddle的同学给出解决方法，谢谢

### PR DESCRIPTION
**Pull request by Yancey1989**
Fixed #6529

### CODE REVIEW

File: python/paddle/v2/reader/tests/decorator_test.py [hunk 1]
```diff
@@ -144,6 +144,36 @@ def mapper(x):
...
+
+        records = [str(i) for i in xrange(5)]
+        temp = tempfile.NamedTemporaryFile()
+        try:
+            with open(temp.name, 'w') as f:
+                for r in records:
+                    f.write('%s\n' % r)
+
+            cmd = "cat %s" % temp.name
+            #reader = paddle.v2.reader.pipe_reader(cmd,
```
Comment (typhoonzero) [inline python/paddle/v2/reader/tests/decorator_test.py]: Remove this commented line please.

File: python/paddle/v2/reader/tests/decorator_test.py [hunk 2]
```diff
@@ -144,6 +144,36 @@ def mapper(x):
...
+            reader = pipe_reader(cmd,
+                                                  simple_parser,
```
Comment (typhoonzero) [inline python/paddle/v2/reader/tests/decorator_test.py]: Indentation error?

File: python/paddle/v2/reader/tests/decorator_test.py [hunk 3]
```diff
@@ -144,6 +144,36 @@ def mapper(x):
...
+                                                  bufsize = 128)
+            # test for 4 passes
```
Comment (typhoonzero) [inline python/paddle/v2/reader/tests/decorator_test.py]: This comment is not needed.

File: python/paddle/v2/reader/tests/decorator_test.py (hunk 2 above)
Comment (Yancey1989) [reply python/paddle/v2/reader/tests/decorator_test.py]: Delete the unused comment and it's just for test.

File: python/paddle/v2/reader/tests/decorator_test.py (hunk 1 above)
Comment (Yancey1989) [reply python/paddle/v2/reader/tests/decorator_test.py]: Done.

**Pull request merged**